eng/pipelines/templates/steps/test-and-publish.yml

```diff
@@ -0,0 +1,16 @@
+steps:
+  - script: dotnet test eng/service.proj --filter TestCategory!=Live --framework ${{ parameters.TFM }} --logger "trx;LogFileName=${{ parameters.Framework}}.trx" /p:ServiceDirectory=${{ parameters.ServiceDirectory }} /p:PackageDirectory=${{ parameters.PackageDirectory }}
+    displayName: "Build & Test (${{ parameters.Framework }})"
+    env:
+      DOTNET_SKIP_FIRST_TIME_EXPERIENCE: 1
+      DOTNET_CLI_TELEMETRY_OPTOUT: 1
+      DOTNET_MULTILEVEL_LOOKUP: 0
+    condition: or( eq(variables['OSName'], 'Windows'), eq( '${{ parameters.Framework }}', 'DotNetCore'))
```

Comment:
You should just remove this condition and add a condition on the template for net461 and exclude that if it is non-windows.

Reply:
Conditions are not working on templates.